### ISSUE TITLE
Is there any way to check if two sentences belong to the same document?

### ISSUE THREAD

**Issue by sidsvash26** [decompositional-semantics-initiative/decomp#2]
Is there any way to check if `uds['ewt-train-1']` and `uds['ewt-train-15']` belong in the same document in the EWT?

Also, how do I resolve queries like this:
what is the id of all other sentences which belong to the same document as `uds['ewt-train-1']`

If this is not currently possible to check, can this feature be added? I think it would be very useful to know the document information.

I can look into adding this feature as a new pull request. Please just point me to the code where changes would need to be made.

**Comment by aaronstevenwhite** [comment]
This is a little tricky because we use UD-EWT v1.2, which does not contain document and sentence IDs (unlike more recent versions of UD-EWT).  We use UD-EWT v1.2 for two reasons: (i) all of the UDS datasets were originally annotated on that version (though AFAIK, the sentences and parses are exactly the same in the same order within each split across versions); and (ii) when Decomp0.1 was designed (and I believe still) PredPatt did not handle newer versions of UD-EWT. The second is a problem because we could in principle use a newer version of UD-EWT that contains document and sentence IDs and associate those IDs with the graph somehow at graph build time graph building within `decomp.syntax.dependency`, but further preprocessing would need to be done before the CoNLL-U was passed to PredPatt (and either way, it would be something of a hack that could come back to bite us in later versions of UD).

One potential patch is to treat the document and sentence ID as an annotation, which could be dropped into the `decomp/data/1.0/annotations/` directory. This would require one to choose which node this information is an annotation of (maybe the syntactic root). This would sidestep the issue of having to preprocess the CoNLL-U before passing it to PredPatt and would allow us to continue using v1.2 (assuming that the document/sentence ID annotation file is constructed from a version of UD containing the same sentences as in v1.2 and this was checked at annotation build time).

The more ambitious patch is to allow for arbitrary versions of UD-EWT to be handled (e.g. whatever the latest release is) and then preprocess the CoNLL-U before passing it to PredPatt (as mentioned above). I'm weary of this option in case there are any major changes to UD-EWT in the next (sub)version, so I would probably prefer continuing to use v1.2 (since that's explicitly what the annotations were done on top of).

In either case, please include properties for `UDSGraph`s named something like `sentenceid` and `documentid` so that the particular backend implementation can be changed later if necessary.
